src/journal.rs

```diff
@@ -30,19 +30,66 @@ where
     unsafe { ffi::sd_journal_sendv(iovecs.as_ptr(), iovecs.len() as c_int) }
 }
 
+fn collect_and_send_result<T, S>(args: T) -> Result<()>
+where
+    T: Iterator<Item = S>,
+    S: AsRef<str>,
+{
+    let iovecs: Vec<const_iovec> = args
+        // SAFETY: we manually guarantee that the lifetime of const_iovec does not exceed that of
+        // the data it's referencing in order to avoid additional allocations.
+        .map(|x| unsafe { const_iovec::from_str(x) })
+        .collect();
+
+    if iovecs.len() > c_int::MAX as usize {
+        return Err(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "Too many iovecs for systemd journal"
+        ));
+    }
```

Comment:
Use `iovecs.len().try_into()` instead of attempting to manually determine if a cast would overflow.